### PR TITLE
Fix image registry

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
#### Summary of Changes (What does this PR do?)
- Sets the image name so that new changes are the ones deployed
- Removes obsolete files that were causing build errors
  image

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?
- [ticket_id]()

#### Screenshots (optional)

### CODE REVIEW

File: .github/workflows/deploy-frontend-pr-previews.yml [hunk 1]
```diff
@@ -118,8 +118,8 @@ jobs:
       - name: Build and Push Image
         run: |
           cd netmanager/
-          docker build --tag ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/pr-previews/netmanager-pr-previews:${{ github.sha }} ./
-          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/pr-previews/netmanager-pr-previews:${{ github.sha }}
+          docker build --tag gcr.io/airqo-250220/pr-previews/netmanager-pr-previews:${{ github.sha }} ./
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontend-pr-previews.yml:121]: Adds the image registry url

File: k8s/docs/values-stage.yaml [hunk 2]
```diff
@@ -1,13 +1,9 @@
...
-# This is a YAML-formatted file.
-# Declare variables to be passed into your templates.
-
 replicaCount: 1
 
 image:
   repository: gcr.io/airqo-250220/airqo-stage-docs
   pullPolicy: Always
-  tag: latest
+  tag: stage-090ad16f-1667922055
```
Comment (123MwanjeMike) [inline k8s/docs/values-stage.yaml:6]: Sets the latest built image

File: .github/workflows/deploy-frontend-pr-previews.yml [hunk 3]
```diff
@@ -118,8 +118,8 @@ jobs:
...
+          docker push gcr.io/airqo-250220/pr-previews/netmanager-pr-previews:${{ github.sha }}
```
Comment (Baalmart) [inline .github/workflows/deploy-frontend-pr-previews.yml:122]: Hi @123MwanjeMike , what happened to using ENV variables? Any challenges?

File: .github/workflows/deploy-frontends-to-production.yml [hunk 4]
```diff
@@ -81,10 +81,10 @@ jobs:
       - name: Build and Push Docker Image
         run: |
           cd netmanager/
-          docker build --tag ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-platform-frontend:${{ needs.image-tag.outputs.build_id }} .
-          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-platform-frontend:${{ needs.image-tag.outputs.build_id }}
-          docker tag ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-platform-frontend:${{ needs.image-tag.outputs.build_id }} ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-platform-frontend:latest
-          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-platform-frontend:latest
+          docker build --tag gcr.io/airqo-250220/airqo-platform-frontend:${{ needs.image-tag.outputs.build_id }} .
+          docker push gcr.io/airqo-250220/airqo-platform-frontend:${{ needs.image-tag.outputs.build_id }}
+          docker tag gcr.io/airqo-250220/airqo-platform-frontend:${{ needs.image-tag.outputs.build_id }} gcr.io/airqo-250220/airqo-platform-frontend:latest
```
Comment (Baalmart) [inline .github/workflows/deploy-frontends-to-production.yml:86]: Same here, is the system failing to pick up our ENVS? Why are we not able to use our secrets? @123MwanjeMike

**Review by Baalmart — COMMENTED** [review]
Hi @123MwanjeMike , this is great work! And many thanks for using the review comments, they make your PRs nice to review. Perfecting marketing stunt! :) 

Otherwise, I have just left a few comments for your review. Thanks again!

File: .github/workflows/deploy-frontend-pr-previews.yml (hunk 3 above)
Comment (123MwanjeMike) [reply .github/workflows/deploy-frontend-pr-previews.yml:122]: Hi @Baalmart, the use of the registry url as an ENV variable was causing several mismatches. It was build and pushing a different image that's why deploys to stage/prod were always having old versions deployed.

Also note that this is not secret information but was rather added as a mechanism to easily change registries with minimal change to the codebase.

File: .github/workflows/deploy-frontends-to-production.yml (hunk 4 above)
Comment (123MwanjeMike) [reply .github/workflows/deploy-frontends-to-production.yml:86]: Yes @Baalmart we are no longer adding them as secrets given it's not secret information. One needs the credentials files to login and access those images plus they are already specified in all our k8s files.

@danielogen can help confirm this.

File: k8s/docs/values-stage.yaml (hunk 2 above)
Comment (NoahNsimbe) [reply k8s/docs/values-stage.yaml:6]: Hi @123MwanjeMike, why this change? The default tag for the latest image is always `latest`

File: .github/workflows/deploy-frontend-pr-previews.yml (hunk 3 above)
Comment (NoahNsimbe) [reply .github/workflows/deploy-frontend-pr-previews.yml:122]: Thanks @123MwanjeMike, Just have a few queries.

`How is it easy to change registries with minimal change to the codebase??....`
Does one have a raise a PR with over 70+ changes??? if so, isn't very easy for a human error to occur while making the changes??


Trying to understand the logic here.
if `secrets.REGISTRY_URL = gcr.io` and `secrets.PROJECT_ID =  airqo-250220`
how is
`${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}` different from `gcr.io/airqo-250220` ?

**Review by NoahNsimbe — COMMENTED** [review]
Thanks @123MwanjeMike ....Just a few comments

File: k8s/docs/values-stage.yaml (hunk 2 above)
Comment (123MwanjeMike) [reply k8s/docs/values-stage.yaml:6]: Yes @NoahNsimbe 
The default image tag (for the k8s manifests) is latest however for ArgoCD to be triggered, it needs to detect a change in the Helm values files thats why we have a step to auto update the image tag in the respective values files.

File: .github/workflows/deploy-frontend-pr-previews.yml (hunk 3 above)
Comment (123MwanjeMike) [reply .github/workflows/deploy-frontend-pr-previews.yml:122]: Hi @NoahNsimbe 
The two are the same actually.
The current issue I am having is that a human error made in the secrets is harder to detect than when it’s made in the visible codebase. That’s why I’m always manually deploying all these applications after changes are not realised.
With regard to the many code changes, I don’t think we shall change our image registry or at least in the foreseeable future.

File: k8s/docs/values-stage.yaml (hunk 2 above)
Comment (Baalmart) [reply k8s/docs/values-stage.yaml:6]: Hi @123MwanjeMike , as the part of "auto update", I am failing to appreciate the "auto" part of this change?

File: .github/workflows/deploy-frontend-pr-previews.yml [hunk 5]
```diff
@@ -5,6 +5,10 @@ on:
     branches:
       - staging
 
+env:
+  REGISTRY_URL: gcr.io
+  PROJECT_ID: airqo-250220
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontend-pr-previews.yml:10]: Adding `REGISTRY_URL` and `PROJECT_ID` as global variables

File: .github/workflows/deploy-frontend-pr-previews.yml [hunk 6]
```diff
@@ -118,8 +122,8 @@ jobs:
       - name: Build and Push Image
         run: |
           cd netmanager/
-          docker build --tag ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/pr-previews/netmanager-pr-previews:${{ github.sha }} ./
-          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/pr-previews/netmanager-pr-previews:${{ github.sha }}
+          docker build --tag ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/pr-previews/netmanager-pr-previews:${{ github.sha }} ./
+          docker push ${{ env.REGISTRY_URL }}/${{ env.PROJECT_ID }}/pr-previews/netmanager-pr-previews:${{ github.sha }}
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontend-pr-previews.yml:126]: Using the globally set values of the `REGISTRY_URL` and `PROJECT_ID` environmental variables.

**Review by 123MwanjeMike — COMMENTED** [review]
Hi @NoahNsimbe and @Baalmart 
I've incorporated the latest feedback on the discussion we had on this PR. Please review again.

File: .github/workflows/deploy-frontend-pr-previews.yml (hunk 6 above)
Comment (Baalmart) [reply .github/workflows/deploy-frontend-pr-previews.yml:126]: Brilliant! Thanks @123MwanjeMike

File: .github/workflows/deploy-frontend-pr-previews.yml (hunk 5 above)
Comment (Baalmart) [reply .github/workflows/deploy-frontend-pr-previews.yml:10]: Awesome, thanks @123MwanjeMike

**Comment by Baalmart** [comment]
Hi @123MwanjeMike , please note that this PR has a couple of merge conflicts that needs to be addressed. Thanks.

**Comment by 123MwanjeMike** [comment]
Thanks @Baalmart 
The conflicts have been resolved accordingly.

**Review by NoahNsimbe — APPROVED** [review]
Thanks @123MwanjeMike 
LGTM

**Review by Baalmart — APPROVED** [review]
Thanks @123MwanjeMike , this is awesome!

Just one comment, I am still curious as we still have different configuration files for staging and environment yet we are using Helm charts.

Perhaps I could just be missing something?

**Comment by 123MwanjeMike** [comment]
> Thanks @123MwanjeMike , this is awesome!
> 
> Just one comment, I am still curious as we still have different configuration files for staging and environment yet we are using Helm charts.
> 
> Perhaps I could just be missing something?

Yes @Baalmart 
Those are the default files of each of the environments.
You'll notice that whereas helm removes the repetition(resultant files are smaller than the k8s files), it does not remove the need to specify values unique to the individual environments.

The template files are the most crucial ones.

**Comment by Baalmart** [comment]
> > Thanks @123MwanjeMike , this is awesome!
> > Just one comment, I am still curious as we still have different configuration files for staging and environment yet we are using Helm charts.
> > Perhaps I could just be missing something?
> 
> Yes @Baalmart Those are the default files of each of the environments. You'll notice that whereas helm removes the repetition(resultant files are smaller than the k8s files), it does not remove the need to specify values unique to the individual environments.
> 
> The template files are the most crucial ones.

Alright @123MwanjeMike !

**Comment by Baalmart** [comment]
Hi @123MwanjeMike , please first resolve thee merge conflicts here. Thanks.

**Review by Baalmart — CHANGES_REQUESTED** [review]
@123MwanjeMike , please resolve the new merge conflicts.

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**